packages/backend-modules/access/lib/grants.js

```diff
@@ -432,13 +465,19 @@ const invalidate = async (grant, reason, t, pgdb, mail) => {
         })
       }
 
+      const perks = await revokePerks(grant, recipient, campaign, pgdb)
+      if (perks.length > 0) {
+        grant.perks = {}
+
+        await Promise.map(perks, (perk) => {
+          eventsLib.log(grant, `perk.${perk.name} revoked`, pgdb)
```

Comment:
```suggestion
          eventsLib.log(grant, `perk.${perk.name}.revoked`, pgdb)
```

(use chaining with dots in event names like `role.member.remove`, `email.recipient.recommendations`, etc.)